### PR TITLE
Simplify FSDP Gradient Clipping

### PR DESCRIPTION
**Pull request by mvpatel2000**
# What does this PR do?

Updates FSDP Gradient Clipping to use root function instead of try/except

### CODE REVIEW

**Review by bcui19 — APPROVED** [review]
LGTM!

File: composer/algorithms/gradient_clipping/gradient_clipping.py [hunk 1]
```diff
@@ -44,38 +43,22 @@ def apply_gradient_clipping(model: Union[ComposerModel, torch.nn.Module], clippi
...
-        _apply_agc(parameters, clipping_threshold=clipping_threshold)
-    elif clipping_type == 'norm':
-        torch.nn.utils.clip_grad_norm_(parameters, max_norm=clipping_threshold)
-    elif clipping_type == 'value':
-        torch.nn.utils.clip_grad_value_(parameters, clip_value=clipping_threshold)
+            if isinstance(module, FullyShardedDataParallel) and module.check_is_root():
+                if clipping_type == 'norm':
+                    module.clip_grad_norm_(max_norm=clipping_threshold)
+                elif clipping_type == 'value':
+                    module.clip_grad_norm_(max_norm=clipping_threshold, norm_type=float('inf'))
```
Comment (vchiley) [inline composer/algorithms/gradient_clipping/gradient_clipping.py:51]:
```suggestion
                    module.clip_grad_norm_(max_norm=clipping_threshold, norm_type=float('inf'))
                else:
                    raise ValueError(f"clipping type must be 'norm' or 'value' with FSDP not {clipping_type}")
```
